extensions/blocks/map/edit.js

```diff
@@ -85,27 +85,31 @@ class MapEdit extends Component {
 	};
 	apiCall( serviceApiKey = null, method = 'GET' ) {
 		const { noticeOperations } = this.props;
-		const { apiKey } = this.state;
 		const path = '/wpcom/v2/service-api-keys/mapbox';
 		const fetch = serviceApiKey
 			? { path, method, data: { service_api_key: serviceApiKey } }
 			: { path, method };
 		this.setState( { apiRequestOutstanding: true }, () => {
 			apiFetch( fetch ).then(
-				result => {
+				( { service_api_key: apiKey, service_api_key_source: apiKeySource } ) => {
```

Comment:
A simpler model for this might be that the endpoint returns up to 2 api keys (aka a list of keys), and the site key is used first (and available to edit in the block settings), with a fallback to the wpcom key for rendering the map (but not shown in the settings).

Reply:
As per [specs](https://github.com/Automattic/Dotcom-roadmap/issues/74), we don't want WPCOM sites to have to provide an access token, so in all cases it's always an either/or kind of situation: "either the site can use the WPCOM token, or they have to provide their own".

As far as I can see, there's only one exception: WPCOM sites that have already provided their own token.
Service API Keys checks for that as first thing, so those sites will continue to use their own token.
Should they remove it, they'd automatically fall back to the WPCOM one.

Reply:
> we don't want WPCOM sites to have to provide an access token

There will likely be a threshold for traffic when we ask users to provide their own key, depending on how rollout of this change goes.